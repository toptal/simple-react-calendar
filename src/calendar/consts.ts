export const BLOCK_CLASS_NAME = 'calendar'
export const NEXT_MONTH_TITLE = 'Next month'
export const PREV_MONTH_TITLE = 'Previous month'
export const DAYS_IN_WEEK = [0, 1, 2, 3, 4, 5, 6]
export const DAYS_OF_WEEK = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
