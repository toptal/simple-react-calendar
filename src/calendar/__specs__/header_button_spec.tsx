import { shallow } from 'enzyme'
import React from 'react'

import { BLOCK_CLASS_NAME, NEXT_MONTH_TITLE, PREV_MONTH_TITLE } from '../consts'
import HeaderButton from '../header_button'

describe('HeaderButton', () => {
  let wrapper, props

  beforeEach(() => {
    props = getProps()
    wrapper = shallow(<HeaderButton {...props} />)
  })

  it('renders <HeaderButton />', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('when prop `enabled` is `false`', () => {
    it('renders <HeaderButton />', () => {
      wrapper.setProps({enabled: false})

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when prop `type` is "next"', () => {
    it('renders <HeaderButton />', () => {
      wrapper.setProps({type: 'next', title: NEXT_MONTH_TITLE})

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when prop `blockClassName` is defined', () => {
    it('renders <HeaderButton />', () => {
      wrapper.setProps({blockClassName: 'cal'})

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when prop `title` is defined', () => {
    it('renders <HeaderButton />', () => {
      wrapper.setProps({title: 'testTitle'})

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when prop `arrow` is defined', () => {
    it('renders <HeaderButton />', () => {
      wrapper.setProps({arrow: <i className="icon">Test</i>})

      expect(wrapper).toMatchSnapshot()
    })
  })
})

const getProps = (overrides = {}) => ({
  type: 'prev',
  enabled: true,
  onClick: () => {},
  blockClassName: BLOCK_CLASS_NAME,
  title: PREV_MONTH_TITLE,
  ...overrides,
})
