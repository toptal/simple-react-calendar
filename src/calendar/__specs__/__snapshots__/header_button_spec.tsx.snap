// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton renders <HeaderButton /> 1`] = `
<button
  className="calendar-header_button is-prev"
  disabled={false}
  onClick={[Function]}
  title="Previous month"
  type="button"
/>
`;

exports[`HeaderButton when prop \`arrow\` is defined renders <HeaderButton /> 1`] = `
<button
  className="calendar-header_button is-prev"
  disabled={false}
  onClick={[Function]}
  title="Previous month"
  type="button"
>
  <i
    className="icon"
  >
    Test
  </i>
</button>
`;

exports[`HeaderButton when prop \`blockClassName\` is defined renders <HeaderButton /> 1`] = `
<button
  className="cal-header_button is-prev"
  disabled={false}
  onClick={[Function]}
  title="Previous month"
  type="button"
/>
`;

exports[`HeaderButton when prop \`enabled\` is \`false\` renders <HeaderButton /> 1`] = `
<button
  className="calendar-header_button is-prev is-disabled"
  disabled={true}
  onClick={[Function]}
  title="Previous month"
  type="button"
/>
`;

exports[`HeaderButton when prop \`title\` is defined renders <HeaderButton /> 1`] = `
<button
  className="calendar-header_button is-prev"
  disabled={false}
  onClick={[Function]}
  title="testTitle"
  type="button"
/>
`;

exports[`HeaderButton when prop \`type\` is "next" renders <HeaderButton /> 1`] = `
<button
  className="calendar-header_button is-next"
  disabled={false}
  onClick={[Function]}
  title="Next month"
  type="button"
/>
`;
